From 43f71cec9953a6104cd4dff5b3d02a964ce09560 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:39 -0400
Subject: [scsi] hpsa: Do not zero fields of ioaccel2 command structure twice

Message-id: <1410186734-19986-49-git-send-email-thenzl@redhat.com>
Patchwork-id: 92083
O-Subject: [RHEL6.6 e-stor PATCH 48/83] hpsa: Do not zero fields of ioaccel2 command structure twice
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Structure was already memset to zero at the top
of hpsa_scsi_ioaccel2_queue_command

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit df03c3f62837e20e936fe705aa480e8bee6bb3ca)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index f546420..1b0db02 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -3588,8 +3588,6 @@ static int hpsa_scsi_ioaccel2_queue_command(struct ctlr_info *h,
  cp->Tag = (c->cmdindex << DIRECT_LOOKUP_SHIFT) |
     DIRECT_LOOKUP_BIT;
  memcpy(cp->cdb, cdb, sizeof(cp->cdb));
- memset(cp->cciss_lun, 0, sizeof(cp->cciss_lun));
- cp->cmd_priority_task_attr = 0;
 
  /* fill in sg elements */
  cp->sg_count = (u8) use_sg;
-- 
1.7.1